External/FEXCore/unittests/Emitter/System_Tests.cpp

```diff
@@ -0,0 +1,147 @@
+#include "TestDisassembler.h"
+
+#include <catch2/catch.hpp>
+#include <fcntl.h>
+
+static TestDisassembler Emit;
+using namespace FEXCore::ARMEmitter;
+
+#define TEST_SINGLE(emit, expected) \
+{ \
+  Emit.emit; \
+  const auto Str = Emit.DisassembleSingle(); \
+  CHECK(Str == expected); \
```

Comment:
On failure, this currently outputs something like:
```
~/FEX/External/FEXCore/unittests/Emitter/ALU_Tests.cpp:771: FAILURE:
  CHECK( Str == "neg x30, x29" )
with expansion:
  "neg x30, x29" == "neg x30, x29"
```

By changing this to `CHECK((emit, DisassembleSingle()) == expected)`, we'd make this a bit more descriptive at the expense of verbosity:
```
~/FEX/External/FEXCore/unittests/Emitter/ALU_Tests.cpp:128: FAILURE:
  CHECK( (sub(Size::i32Bit, Reg::r29, Reg::r28, 0, true), DisassembleSingle()) == "sub w29, w28, #0x0 (0)" )
with expansion:
  "sub w29, w28, #0x0 (0)"
  ==
  "sub w29, w28, #0x0 (0)"
```

Not sure it's a better trade-off, but maybe worth considering :)

Reply:
Changed it over to that. Resolved.